packages/orchestrator-ui-components/src/components/WFOSubscription/WFOSubscription.tsx

```diff
@@ -1,121 +1,44 @@
-import React, { FC, useContext, useState } from 'react';
+import React, { useState } from 'react';
+import { useTranslations } from 'next-intl';
 import {
-    EuiBadge,
     EuiFlexGroup,
     EuiFlexItem,
     EuiTab,
     EuiTabs,
     EuiText,
 } from '@elastic/eui';
-import { useQuery } from 'react-query';
-import { GraphQLClient } from 'graphql-request';
-import { useTranslations } from 'next-intl';
+import { useQueryWithGraphql } from '../../hooks';
 
-import { OrchestratorConfigContext, SubscriptionContext } from '../../contexts';
-import {
-    GENERAL_TAB,
-    getColor,
-    PROCESSES_TAB,
-    SERVICE_CONFIGURATION_TAB,
-    tabs,
-} from './utils';
-import {
-    getSubscriptionsDetailCompleteGraphQlQuery,
-    getSubscriptionsDetailOutlineGraphQlQuery,
-    mapApiResponseToSubscriptionDetail,
-} from '../../graphqlQueries';
-import { WFOSubscriptionActions } from './WFOSubscriptionActions';
-import { WFOSubscriptionGeneral } from './WFOSubscriptionGeneral';
-import { WFOSubscriptionDetailTree } from './WFOSubscriptionDetailTree';
+import { SubscriptionTabIds, tabs } from './utils';
+
+import { GET_SUBSCRIPTION_DETAIL_GRAPHQL_QUERY } from '../../graphqlQueries';
+import { WFOLoading } from '../WFOLoading';
+import { SubscriptionActions } from './SubscriptionActions';
+import { SubscriptionGeneral } from './SubscriptionGeneral';
+import { SubscriptionDetailTree } from './SubscriptionDetailTree';
 import { ProcessesTimeline } from './WFOProcessesTimeline';
 
 type WFOSubscriptionProps = {
     subscriptionId: string;
 };
 
-export const WFOSubscription: FC<WFOSubscriptionProps> = ({
-    subscriptionId,
-}) => {
+export const WFOSubscription = ({ subscriptionId }: WFOSubscriptionProps) => {
     const t = useTranslations('subscriptions.detail');
-    const { graphqlEndpointCore } = useContext(OrchestratorConfigContext);
-    const { subscriptionData, setSubscriptionData, loadingStatus } =
-        React.useContext(SubscriptionContext);
+    const [selectedTabId, setSelectedTabId] = useState<SubscriptionTabIds>(
+        SubscriptionTabIds.GENERAL_TAB,
+    );
 
-    const [selectedTabId, setSelectedTabId] = useState(GENERAL_TAB);
+    const { data, isFetching } = useQueryWithGraphql(
+        GET_SUBSCRIPTION_DETAIL_GRAPHQL_QUERY,
+        { subscriptionId },
+        `subscription-${subscriptionId}`,
+        true,
+    );
 
-    const onSelectedTabChanged = (id: string) => {
+    const onSelectedTabChanged = (id: SubscriptionTabIds) => {
         setSelectedTabId(id);
     };
 
-    // Todo #97: Find out if pre fetch can be used again. The shape of table cache seems to have changed
```

Comment:
This Todo; is still valid.

We had a working cache when navigationg from table to detail page.
Removing it also removes the link to https://github.com/workfloworchestrator/orchestrator-ui/issues/97

Reply:
The issue was a little unclear to me: https://github.com/workfloworchestrator/orchestrator-ui/issues/97 I can put the TODO back but I'm not sure what to do when implementing the ticket

Reply:
Readded the TODO. Can you clarify the ticket a bit?